pyuvdata/uvbeam.py

```diff
@@ -1817,34 +1815,73 @@ def write_beamfits(self, filename, run_check=True, check_extra=True,
                                     clobber=clobber)
         del(beamfits_obj)
 
-    def read_cst_beam(self, filename, beam_type='power', feed_pol='x', rotate_pol=None,
+    def _read_cst_beam_yaml(self, filename):
+        import yaml
+
+        with open(filename, 'r') as file:
+            settings_dict = yaml.safe_load(file)
+
+        required_keys = ['telescope_name', 'feed_name', 'feed_version',
+                         'model_name', 'model_version', 'history', 'frequencies',
+                         'filenames', 'feed_pol']
+
+        for key in required_keys:
+            if key not in settings_dict:
+                raise ValueError('{key} is a required key in CST settings files '
+                                 'but is not present.'.format(key=key))
+
+        return settings_dict
+
+    def read_cst_beam(self, filename, beam_type='power', feed_pol=None, rotate_pol=None,
                       frequency=None, telescope_name=None, feed_name=None,
                       feed_version=None, model_name=None, model_version=None,
-                      history='', run_check=True, check_extra=True,
+                      history=None, reference_impedance=None, extra_keywords=None,
+                      frequency_select=None, run_check=True, check_extra=True,
                       run_check_acceptability=True):
         """
         Read in data from a cst file.
 
         Args:
-            filename: The cst file or list of files to read from. If a list is passed,
+            filename (str): Either a settings yaml file or a cst text file or
+                list of cst text files to read from. If a list is passed,
                 the files are combined along the appropriate axes.
-            beam_type: what beam_type to read in ('power' or 'efield'). Defaults to 'power'.
-            feed_pol: the feed or polarization or list of feeds or polarizations the files correspond to.
+                Settings yaml files must include the following keywords:
+                    |  telescope_name (str)
+                    |  feed_name (str)
+                    |  feed_version (str)
+                    |  model_name (str)
+                    |  model_version (str)
+                    |  history (str)
+                    |  frequencies (list(float))
+                    |  cst text filenames (list(str)) -- path relative to yaml file location
+                    |  feed_pol (str) or (list(str))
+                and they may include the following optional keywords:
+                    | ref_imp (float): beam model reference impedance
+                    | all other fields will go into the extra_keywords attribute
+                Specifying any of the associated keywords to this function will
```

Comment:
I think this is a good practice, but do we want to warn/alert the users if an override happens? I'm not familiar with the day-to-day use of the beams, so I'm not sure how common it is. If users frequently override what's in the file, then it'd probably be annoying to get lots of warnings.

Reply:
I think it'd be pretty unusual to want to override this values. I can add warnings.

Reply:
done.